test/functional/rpc_fundrawtransaction.py

```diff
@@ -1200,6 +1201,67 @@ def test_add_inputs_default_value(self):
 
         self.nodes[2].unloadwallet("test_preset_inputs")
 
+    def test_preset_inputs_selection_crash(self):
+        self.log.info('Test wallet preset inputs selection')
+
+        # This exercises the bug presented in #26559 that
+        # ended up on (1) a wallet crash or (2) a created and
+        # broadcasted tx that pays an insane fee amount.
+        #
+        # This is covered by making the wallet selects the preset
+        # inputs twice during the coin selection process (which
+        # is a dangerous bug on v24.0).
+        #
+        # Making the wallet think that the selection process result covers
+        # the entire tx target when it does not. It's actually creating
+        # a tx that sends more coins than what inputs are covering for.
+        #
+        # Which, combined with the subtract fee from outputs option,
+        # makes the wallet generate and broadcast a tx that pays
+        # up to the sum of all the preset inputs, minus one, amounts in
+        # fee :(.
+
+        # Create and fund the wallet with 2 UTXO of 5 BTC each
+        self.nodes[2].createwallet("test_preset_inputs_selection")
+        wallet = self.nodes[2].get_wallet_rpc("test_preset_inputs_selection")
+        outputs = {}
+        for _ in range(2):
+            outputs[wallet.getnewaddress(address_type="bech32")] = 5
+        self.nodes[0].sendmany("", outputs)
+        self.generate(self.nodes[0], 1)
+
+        # Select the preset inputs
+        coins = wallet.listunspent()
+        preset_inputs = [coins[0], coins[1]]
+
+        # Now let's create the tx
+        options = {
+            "inputs": [
+                {
+                    "txid": preset_inputs[0]["txid"],
+                    "vout": preset_inputs[0]["vout"]
+                },
+                {
+                    "txid": preset_inputs[1]["txid"],
+                    "vout": preset_inputs[1]["vout"]
+                },
+            ],
+            "add_inputs": True,
+            "subtract_fee_from_outputs": [0],
+            "add_to_wallet": False
+        }
+
+        # First case, use 'subtract_fee_from_outputs' to make the wallet create a tx that pays an insane fee on v24.0.
+        # (further details in #26559).
```

Comment:
There is no need to continually write `v24.0` and `(further details in #26559)` (here and below and further above). If you need to provide further context or details, please put them in the commit (or PR) descriptions, so they are available with the code change.

Reply:
those were my initial thought actually, added this on the last push per request https://github.com/bitcoin/bitcoin/pull/26560#discussion_r1034800106.

seems that not everyone read the PR/commit description. But well, will just revert it.

Reply:
Sorry if this wasn't clear in my comment, but I wasn't suggesting adding `v24` and links to the PR. My comment was to point out that the comments in the test didn't match what was being checked in the assertion, which would be confusing to someone who didn't have the background context. I understand this is a bit tricky because the test is for a bug that is no longer there, but ideally, the comment should explain why you are asserting "Insufficient Funds" in the happy case, and why, if the bug were present, that you would expect the "Insufficient Funds" assertion to fail. If you do want to link to the PR for more context, perhaps a top-level comment would be the best place?

Reply:
Ok, done. Ready for another review round.